spatialmath/dualquaternion.go

```diff
@@ -185,6 +186,39 @@ func (q *dualQuaternion) Transformation(by dualquat.Number) dualquat.Number {
 	}
 }
 
+// Format implements fmt.Formatter to allow for finer grain control over printing.
+// dualquat.Number also has this implemented so in order to get custom printing we need to also implement this as opposed to simply
+// implementing fmt.Stringer.
+// Some example verbs and how they would display a zero pose are shown below
+//
+//	Verb    | String
+//	%v		| {X:0 Y:0 Z:0 OX:0 OY:0 OZ:1 Theta:0°}
+//	%.3v	| {X:0.000 Y:0.000 Z:0.000 OX:0.000 OY:0.000 OZ:1.000 Theta:0.000°}
+//	%#v		| r3.Vector{0 0 0} *spatialmath.OrientationVectorDegrees{0 0 0 1}
+//	%g		| ((1+0i+0j+0k)+(+0+0i+0j+0k)ϵ)
+func (q *dualQuaternion) Format(fs fmt.State, verb rune) {
+	prec, _ := fs.Precision()
+	width, _ := fs.Width()
+	format := fmt.Sprintf("%%%d.%df", width, prec)
+	pt := q.Point()
+	o := q.Orientation().OrientationVectorDegrees()
+	switch verb {
+	case 'v':
+		if fs.Flag('#') {
```

Comment:
Is `#` a special character here or arbitrary?

Reply:
Its a special character that prints out the type if you do `%#v`.  Adding support for it was pretty easy and I thought possibly useful down the line

Reply:
Gotcha, that makes sense